R-package/demo/interaction_constraints.R

```diff
@@ -0,0 +1,94 @@
+library(xgboost)
+library(data.table)
+
+set.seed(1024)
+
+# Function to obtain a list of interactions fitted in trees
+treeInteractions <- function(in.tree, in.depth){
```

Comment:
On R code style: 
- have spaces after commas and around infix operators
- use TRUE/FALSE instead of T/F
- do not use ambiguous names (does "in" in "in.depth" stand for input? interaction?); do not overuse temp as a name or a part of a name 
- the loose convention within this package is to use underscores in variable names and dots in function names